cmd/dep/init.go

```diff
@@ -209,6 +227,45 @@ func hasImportPathPrefix(s, prefix string) bool {
 	return strings.HasPrefix(s, prefix+"/")
 }
 
+// getVersionConstituents extracts version constituents
+func getVersionConstituents(v gps.Version) (version, revision string) {
+	switch tv := v.(type) {
+	case gps.UnpairedVersion:
+		version = tv.String()
+	case gps.Revision:
+		revision = tv.String()
+	case gps.PairedVersion:
+		version = tv.Unpair().String()
+		revision = tv.Underlying().String()
+	}
+
+	return version, revision
+}
+
+// getProjectPropertiesFromVersion takes a gps.Version and returns a proper
+// gps.ProjectProperties with Constraint value based on the provided version.
+func getProjectPropertiesFromVersion(v gps.Version) gps.ProjectProperties {
+	pp := gps.ProjectProperties{}
+
+	// constituent version and revision. Ignoring revison for manifest.
+	cv, _ := getVersionConstituents(v)
+
+	switch v.Type() {
+	case gps.IsBranch:
+		pp.Constraint = gps.NewBranch(cv)
+	case gps.IsVersion:
+		pp.Constraint = gps.NewVersion(cv)
+	case gps.IsSemver:
+		// TODO: remove "^" when https://github.com/golang/dep/issues/225 is ready.
+		c, _ := gps.NewSemverConstraint("^" + cv)
```

Comment:
Let's check the error here and panic if we get one. That shouldn't be possible - any valid plain semver version should be able to have the caret operand prepended to it - but if we somehow missed something, then program behavior is undefined; panicking is the safe thing to do.